### DISCUSSION TITLE
List output iteration

### DISCUSSION THREAD

**Discussion by lexiismadd** [StackStorm/st2#6214] [Workflows]
Is there a way to iterate over the list output from an action, but performing multiple actions on each list item.

so for example, the result from Action A is a list.
I was Action B, Action C, and Action D to iterate over each list item.

Then maybe the output from Action D is another list, and I want Action E and Action F to iterate over the output from Action D (which is still in the loop iterating over the output of Action A)

Is this possible?

**Comment by nzlosh** [comment]
Yes. You have to use additional workflows to group the actions.  Workflow1 would call ActionA and Workflow2.  Workflow2 would group B, C, D and Workflow3.  Workflow3 would group E and F.  Execution would be conceptually like the below graph:

```
Workflow1
  -> ActionA [output: list1]
  -> Workflow2 [with-items: list1]
    -> ActionB
    -> ActionC
    -> ActionD [output: list2]
    -> Workflow3 [with-items: list2]
      -> ActionE
      -> ActionF
```
Keep in mind Orquesta `with-items` does not behave like common programming loops.  It only applies to a single action or workflow.  The iterated value is only available as a variable to the invoked action using `item()` and is not available in the `next:` section of the action definition.  The `results` keyword will contain the list of results returned by the execution of each action.
